### PR TITLE
#2133 Component Port Allocation is not visible in diagram LAB

### LINKED ISSUE #2133: Component Port Allocation is not visible in diagram LAB, ports are missing
- [ ] Port allocations not displayed for inner (3rd level)

- [ ] When selecting Logical System, the PL accelerator doesn't work

- [ ] PL towards a regular LogicalComponent shall not be created (because is displayed as computed)

### PR DESCRIPTION
**Pull request by MalinaStefaniaStoicanescu**
Resolve #2133 
Change-Id: Ibb27c05556bf83547cd929c91b178ef8c1a847fb
Signed-off-by: MalinaStefaniaStoicanescu <malina.stoicanescu@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.model.helpers/src/org/polarsys/capella/core/model/helpers/PhysicalLinkExt.java [hunk 1]
```diff
@@ -335,7 +335,8 @@ protected static PhysicalPort getPhysicalPortFrom(PhysicalLink pLink, Informatio
       if (pp != null) {
         EObject ppOwner = pp.eContainer();
         if (ppOwner instanceof Component) {
-          List<Component> allSubUsedAndDeployedComponents = ComponentExt.getSubUsedAndDeployedComponents((Component) ppOwner);
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.model.helpers/src/org/polarsys/capella/core/model/helpers/PhysicalLinkExt.java:338]: Can replace only with: getAllSubUsedAndDeployedComponents

File: core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java [hunk 2]
```diff
@@ -160,42 +164,72 @@ protected void createPhysicalLinksFromCExchanges(Component sourceContainer, Comp
...
+
+    if (targetContained instanceof LogicalComponent
+        && (!ComponentExt.isActor(targetContained) && !isSystem((Component) targetContained)))
+      return null;
+
+    return (Component) targetContained;
+  }
+  
+  private boolean isSystem(Component component) {
+    if (component instanceof PhysicalComponent)
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java:225]: Why exclude PA level?

File: core/plugins/org.polarsys.capella.core.projection.exchanges/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -12,7 +12,8 @@ Require-Bundle: org.eclipse.core.runtime,
  org.polarsys.capella.common.ui,
  org.eclipse.sirius.diagram,
  org.polarsys.capella.core.sirius.ui,
- org.polarsys.capella.core.platform.sirius.ui.actions
+ org.polarsys.capella.core.platform.sirius.ui.actions,
+ org.polarsys.capella.core.data.gen
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.projection.exchanges/META-INF/MANIFEST.MF:16]: useful?

File: core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java [hunk 4]
```diff
@@ -160,42 +165,70 @@ protected void createPhysicalLinksFromCExchanges(Component sourceContainer, Comp
     }
   }
   
+  private boolean isContainedByALogicalComponent(Component component) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java:168]: Can you add some comments to explain what it does? 
return true if the given component/actor is owned by a not actor component right ?

File: core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java [hunk 5]
```diff
@@ -160,42 +165,70 @@ protected void createPhysicalLinksFromCExchanges(Component sourceContainer, Comp
...
+        if (!isContainedByALogicalComponent(targetContainer))
+          doCreatePhysicalLink(componentExchange, sourceContainer, targetContainer, port);
       }
     }
   }
   
-  private PhysicalComponent findDeployingComponent(Component sourceContainer, ComponentExchange componentExchange,
-      PhysicalComponent targetContained) {
-    for (Part partition : ((PhysicalComponent) targetContained).getRepresentingParts()) {
+  private Component findTheTargetNode(ComponentExchange componentExchange, ComponentPort port) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java:206]: can you explain the rules in a comment ?
thanks

**Review by pdulth — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java (hunk 4 above)
Comment (MalinaStefaniaStoicanescu) [reply core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java:168]: yes :)

File: core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java (hunk 5 above)
Comment (MalinaStefaniaStoicanescu) [reply core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java:206]: sure

File: core/plugins/org.polarsys.capella.core.projection.exchanges/META-INF/MANIFEST.MF (hunk 3 above)
Comment (MalinaStefaniaStoicanescu) [reply core/plugins/org.polarsys.capella.core.projection.exchanges/META-INF/MANIFEST.MF]: no, deleted

**Review by pdulth — APPROVED** [review]

**Pull request merged**